pkg/controller/shared_state.go

```diff
@@ -1571,16 +1578,18 @@ func (c *SharedState) createAzVolume(ctx context.Context, desiredAzVolume *azdis
 		}
 
 		updated = azVolume.DeepCopy()
-		if azVolume, err = c.azClient.DiskV1beta2().AzVolumes(c.config.ObjectNamespace).Update(ctx, updated, metav1.UpdateOptions{}); err != nil {
+		azVolume, err = c.azClient.DiskV1beta2().AzVolumes(c.config.ObjectNamespace).Update(ctx, updated, metav1.UpdateOptions{})
+		if err != nil {
 			return err
 		}
+		originalAzVolume = azVolume.DeepCopy()
 
 		azVolume.Status.Annotations = azureutils.AddToMap(azVolume.Status.Annotations, statusAnnotation...)
```

Comment:
Any updates to the target object *must* be done within the update function passed to `UpdateCRIWithRetry`. Otherwise, the update will be lost of the initial update attempt fails due to a conflict or other error.

Reply:
Also since you now pass `originalAzVolume` instead of `updated`, this change will not be applied.

Reply:
NM. I see that `updated` is used to carry the changes *into* the update function.

Reply:
If you make the deep copy of `azVolume` to `updated` and then update the status annotations of `updated`, you wouldn't need the `originalAzVolume` copy also.